package.json

```diff
@@ -12,7 +12,7 @@
     "doc": "documentation build lib/** -f md -o docs/api.md",
     "postdoc": "mkdocs build",
     ":clean:doc": "git clean -fdX ./site ./docs",
-    "lint": "eslint .",
+    "lint": "eslint . --ext=.js --ext=.json",
```

Comment:
is there not a way to do this from the eslintrc?  Keeping the command line arguments to a minimum helps if a more targeted call is desired.

Reply:
Pretty sure that the CLI is the one-true-god.

https://eslint.org/docs/user-guide/configuring#specifying-file-extensions-to-lint

> **Specifying File extensions to Lint**
> Currently the sole method for telling ESLint which file extensions to lint is by specifying a comma separated list of extensions using the `--ext` command line option. Note this flag only takes effect in conjunction with directories, and will be ignored if used with filenames or glob patterns.

Reply:
Actually, re-reading that doc comment from above, it seems we could do comma separated, which shortens it [barely]:

```json
"lint": "eslint . --ext=.js,.json",
```

Reply:
the length of the command is less concerning; more worried about doing something like:

```bash
npm run lint lib/datastore.js
```

doing something unexpected.

Reply:
I think we could probably do something like that, if you wanted. May depend on _why_ you want to do that though. If you just want to lint specific files before you commit them, there are probably better solutions though. Usually we'd use [**husky**](http://npm.im/husky) and [**lint-staged**](http://npm.im/lint-staged) or something.

But this may also be a possible solution:

```diff
diff --git a/package.json b/package.json
index 5e8287a..4a8b2f1 100644
--- a/package.json
+++ b/package.json
@@ -12,7 +12,8 @@
     "doc": "documentation build lib/** -f md -o docs/api.md",
     "postdoc": "mkdocs build",
     ":clean:doc": "git clean -fdX ./site ./docs",
-    "lint": "eslint .",
+    "lint": "npm run eslint -- .",
+    "eslint": "eslint --ext=.js --ext=.json",
     "pretest": "npm run lint && npm run :clean:coverage",
     ":test:karma": "karma start",
     ":test:node": "mocha test",
```

1. `npm run lint` would lint _all_ files.
2. `npm run eslint -- lib/datastore.js` would only lint that one file.

Reply:
Note that `npm` no longer requires `--`.

Reply:
Also, thanks for the tool hints, but sometimes I really do want to get the lint issues for a single file, not all the staged changes, not all the unstaged changes.